lib/agenda/create.ts

```diff
@@ -16,6 +16,7 @@ export const create = function (this: Agenda, name: string, data: any): Job {
   const priority = this._definitions[name]
     ? this._definitions[name].priority
     : 0;
-  const job = new Job({ name, data, type: "normal", priority, agenda: this });
+  const saveResult = this._definitions[name] ? this._definitions[name].saveResult || false : false
```

Comment:
Renaming `saveResult` to `shouldSaveResult` will help conveying that its a boolean.